src/metadata.rs

```diff
@@ -24,77 +24,19 @@ static PATH_ILLEGAL_COMPONENTS: &'static [&str] = &[
 
 static PATH_ILLEGAL_COMPONENTS_CASE_INSENSITIVE: &'static [&str] = &[
     // DOS device files
-    "CON",
-    "PRN",
-    "AUX",
-    "NUL",
-    "COM1",
-    "COM2",
-    "COM3",
-    "COM4",
-    "COM5",
-    "COM6",
-    "COM7",
-    "COM8",
-    "COM9",
-    "LPT1",
-    "LPT2",
-    "LPT3",
-    "LPT4",
-    "LPT5",
-    "LPT6",
-    "LPT7",
-    "LPT8",
-    "LPT9",
-    "KEYBD$",
-    "CLOCK$",
-    "SCREEN$",
-    "$IDLE$",
-    "CONFIG$",
+    "CON", "PRN", "AUX", "NUL", "COM1", "COM2", "COM3", "COM4", "COM5",
+    "COM6", "COM7", "COM8", "COM9", "LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8",
+    "LPT9", "KEYBD$", "CLOCK$", "SCREEN$", "$IDLE$", "CONFIG$",
 ];
 
 static PATH_ILLEGAL_STRINGS: &'static [&str] = &[
-    ":", // for *nix compatibility
+    ":",  // for *nix compatibility
     "\\", // for windows compatibility
-    "<",
-    ">",
-    "\"",
-    "|",
-    "?",
-    "*",
-    // control characters, all illegal in FAT
-    "\u{000}",
-    "\u{001}",
-    "\u{002}",
-    "\u{003}",
-    "\u{004}",
-    "\u{005}",
-    "\u{006}",
-    "\u{007}",
-    "\u{008}",
-    "\u{009}",
-    "\u{00a}",
-    "\u{00b}",
-    "\u{00c}",
-    "\u{00d}",
-    "\u{00e}",
-    "\u{00f}",
-    "\u{010}",
-    "\u{011}",
-    "\u{012}",
-    "\u{013}",
-    "\u{014}",
-    "\u{015}",
-    "\u{016}",
-    "\u{017}",
-    "\u{018}",
-    "\u{019}",
-    "\u{01a}",
-    "\u{01b}",
-    "\u{01c}",
-    "\u{01d}",
-    "\u{01e}",
-    "\u{01f}",
+    "<", ">", "\"", "|", "?", "*", // control characters, all illegal in FAT
```

Comment:
Can you reformat these by hand to put the comments over the items they point to. The FAT one needs to drop down a line, and the others should reformatted for consistency.

Reply:
done, turns out it's easy to get rustfmt to ignore this with `#[cfg_attr(rustfmt, rustfmt_skip)]`.